src/Microsoft.AspNet.OData.Shared/Query/DefaultSkipTokenHandler.cs

```diff
@@ -0,0 +1,387 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Linq.Expressions;
+
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Formatter;
+using Microsoft.AspNet.OData.Formatter.Serialization;
+using Microsoft.AspNet.OData.Query.Expressions;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData.Query
+{
+    /// <summary>
+    /// Default implementation of SkipTokenHandler for the service. 
+    /// </summary>
+    public class DefaultSkipTokenHandler : SkipTokenHandler
+    {
+        private const char CommaDelimiter = ',';
+        private static char propertyDelimiter = ':';
```

Comment:
>':' [](start = 48, length = 3)

I am still worry about ':' in the query option. See 

[https://github.com/OData/odata.net/pull/1457](https://github.com/OData/odata.net/pull/1457)
@Michael Pizzo

Reply:
As discussed, will wait for what Mike has to say but I believe my approach is pretty safe as I split on the first : and the property name cannot have a : in it. 

---
In reply to: [282620659](https://github.com/OData/WebApi/pull/1724#discussion_r282620659) [](ancestors = 282620659)